src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -1805,54 +1909,78 @@ namespace rascal {
 
               // Each Cartesian gradient component occupies a contiguous block
               // (row-major storage)
+              // grad_i c^{ib} = - \sum_{j} grad_j c^{ij}
               gradient_center_by_type.block(
                   cartesian_idx * max_radial, l_block_idx,
                   max_radial, l_block_size) -= pair_gradient_contribution;
+              // grad_i c^{ja} = (-1)^{l+1} grad_j c^{ij}
               gradient_neigh_by_type.block(
                   cartesian_idx * max_radial, l_block_idx,
-                  max_radial, l_block_size) += pair_gradient_contribution;
+                  max_radial, l_block_size) = parity *
+                    pair_gradient_contribution;
               l_block_idx += l_block_size;
+              parity *= -1.;
               // clang-format on
             }  // for (angular_l)
           }    // for cartesian_idx
 
-          // half list branch for computing grad_j c^{j} using
-          // grad_j c^{ji} = (-1)^{l} grad_j c^{ij}
+          // half list branch for accumulating parts of grad_j c^{j} using
+          // grad_j c^{ji a} = - grad_i c^{ji a}
           if (IsHalfNL) {
             if (is_center_atom) {
+              // grad_j c^{j}
               auto & coefficients_neigh_center_gradient =
                   expansions_coefficients_gradient[neigh.get_atom_jj()];
```

Comment:
Is this all that was needed to get the half NL accumulation to work?

Reply:
yes because I throw an error when `IsHalfNL == true` and `(2*cutoff > cell_lengths).any()` which ensures that no center can have in its neighborhood another center and a periodic image associated to it.